modules/python/src2/cv2.cpp

```diff
@@ -343,10 +344,21 @@ static bool pyopencv_to(PyObject* o, Mat& m, const ArgInfo info)
         _strides = PyArray_STRIDES(oarr);
     }
 
-    for(int i = 0; i < ndims; i++)
+    // Normalize strides in case NPY_RELAXED_STRIDES is set
+    size_t default_step = elemsize;
+    for ( int i = ndims - 1; i >= 0; --i )
     {
         size[i] = (int)_sizes[i];
-        step[i] = (size_t)_strides[i];
+        if ( size[i] > 1 )
+        {
```

Comment:
I don't know if that is another error in OpenCV, but the current `needcopy` logic doesn't exclude a non-contiguous array, as it only requires non-increasing strides, so for something like:

```
>>> a = np.empty((10, 10, 3), dtype=np.int32)
>>> a = a[::2, ::10]
>>> a.strides
(240, 120, 4)
>>> a.shape
(5, 1, 3)
```

I don't think you would have a copy made, so the original array strides have to be kept unchanged.


```
a = np.empty((10, 1, 2), dtype=np.int